common/src/auth/ServiceConnectionString.cs

```diff
@@ -0,0 +1,156 @@
+    /// </code>
+    ///
+    /// This object parse and store the connection string. It is responsible to provide the authorization token too.
+    /// </remarks>
+    public class ServiceConnectionString
+    {
+        protected static readonly TimeSpan DefaultTokenTimeToLive = TimeSpan.FromMinutes(5);
+
+        /// <summary>
+        /// CONSTRUCOR
+        /// </summary>
+        /// <param name="parser">the <see cref="ServiceConnectionStringParser"/> with the connection string content.</param>
+        /// <exception cref="ArgumentNullException">if the provided parser is null.</exception>
+        public ServiceConnectionString(ServiceConnectionStringParser parser)
+        {
+            if (parser == null)
+            {
+                throw new ArgumentNullException(nameof(parser));
+            }
+
+            HostName = parser.HostName;
+            SharedAccessKeyName = parser.SharedAccessKeyName;
+            SharedAccessKey = parser.SharedAccessKey;
+            SharedAccessSignature = parser.SharedAccessSignatureString;
+            ServiceName = parser.ServiceName;
+            HttpsEndpoint = new UriBuilder("https", parser.HostName).Uri;
+        }
+
+        /// <summary>
+        /// The Service Name
+        /// </summary>
+        public string ServiceName
+        {
+            get;
+            private set;
+        }
+
+        /// <summary>
+        /// The Service Client Hostname
+        /// </summary>
+        public string HostName
+        {
+            get;
+            private set;
+        }
+
+        /// <summary>
+        /// The Service Client Https Endpoint
+        /// </summary>
+        public Uri HttpsEndpoint
+        {
+            get;
+            private set;
+        }
+
+        /// <summary>
+        /// The Service Audience
+        /// </summary>
+        public string Audience
+        {
+            get { return HostName; }
+        }
+
+        /// <summary>
+        /// The Service Access Policy Name
+        /// </summary>
+        public string SharedAccessKeyName
+        {
+            get;
+            private set;
+        }
+
+        /// <summary>
+        /// The Service Shared Access Key for the specified
+        /// access policy
+        /// </summary>
+        public string SharedAccessKey
+        {
+            get;
+            private set;
+        }
+
+        /// <summary>
+        /// The Service Shared Access Signature
+        /// </summary>
+        public string SharedAccessSignature
+        {
+            get;
+            private set;
+        }
+
+        /// <summary>
+        /// Returns the shared access signature for authorization
+        /// </summary>
+        /// <returns></returns>
+        public string GetSasToken()
+        {
+            string password;
+            if (string.IsNullOrWhiteSpace(SharedAccessSignature))
+            {
+                TimeSpan timeToLive;
+                password = BuildToken(out timeToLive);
```

Comment:
Can you do this?

password = BuildToken(out TimeSpan timeToLive);

Reply:
sure